roles/ovirt-engine-setup/tasks/main.yml

```diff
@@ -0,0 +1,72 @@
+---
+# health page
+- block:
+  - name: check if ovirt-engine running (health page)
+    uri:
+      url: "http://{{ ansible_fqdn }}/ovirt-engine/services/health"
+      status_code: 200
+    register: ovirt_engine_status
+    retries: 2
+    delay: 5
+    until: ovirt_engine_status|success
+    ignore_errors: True
+
+  # copy default answer file
+  - name: copy default answerfile
+    template:
+      src: answerfile_{{ ovirt_engine_version }}_basic.txt.j2
+      dest: /tmp/answerfile.txt
+      mode: 0644
+      owner: root
+      group: root
+    when: ovirt_engine_answer_file_path is undefined
+
+  # copy custom answer file
+  - name: copy custom answer file
+    no_log: True
+    template:
+      src: "{{ ovirt_engine_answer_file_path }}"
+      dest: /tmp/answerfile.txt
+      mode: 0600
+      owner: root
+      group: root
+    when: ovirt_engine_answer_file_path is defined
+
+  - name: update setup packages
+    yum:
+      name: 'ovirt*setup*'
+      state: latest
+    when: ovirt_engine_update
+    tags:
+      - skip_ansible_lint
+
+  - name: run engine-setup with answerfile
+    shell: 'engine-setup --config-append=/tmp/answerfile.txt'
+    when: ovirt_engine_status|failed or ovirt_engine_update
+    tags:
+      - skip_ansible_lint
+
+  - name: check state of engine
+    service:
+      name: ovirt-engine
+      state: started
+
+  - name: restart of ovirt-engine service
+    service:
+      name: ovirt-engine
+      state: restarted
```

Comment:
Hmm, if you restart it here, then the `started` state before is useless, no?

Reply:
Why do we restart ovirt-engine there? The service is (re)started during engine-setup, no?

Reply:
It is.